Object commands WIP

svdgoor opened the pull request:
Includes command patches & optimizations
Includes a refactor (nothing changed) in the VolmitSender for sending the commands to the user
jesus that thing was a hell to rewrite (it's better split up now to be more accessible)

svdgoor committed:
Matter & save

svdgoor committed:
Merge remote-tracking branch 'upstream/master' into DecreeCommands

svdgoor committed:
Merge remote-tracking branch 'upstream/master' into DecreeCommands

svdgoor committed:
See description

- No-parameter command nodes are now on-click-runnable
- Parameters are now sorted by: name *and then* required

svdgoor committed:
Aliases, colors & patches

svdgoor committed:
patch debug

svdgoor committed:
Dont sort subvalues to allow better control

svdgoor committed:
Improve boolean handling in DecIris

svdgoor committed:
Don't sort values again (done in getParams)

svdgoor committed:
Engine check

svdgoor committed:
Fuck that builder. Yeet

cyberpwnn commented on src/main/java/com/volmit/iris/core/decrees/DecIris.java:
```diff
@@ -46,9 +50,9 @@
     private DecObject object;
 
-    @Decree(description = "Create a new world", aliases = "+")
+    @Decree(description = "Create a new world", aliases = {"+", "c"})
```
Remove "c", its not needed

svdgoor replied:
It is, otherwise it defaults to obje***c***ts. No joke.

cyberpwnn commented on src/main/java/com/volmit/iris/core/decrees/DecIris.java:
```diff
@@ -127,18 +131,31 @@ public void bitwise(
-            @Param(name = "on", description = "Whether or not debug should be on", defaultValue = "true")
-                    boolean on
+            @Param(name = "on", description = "Whether or not debug should be on", defaultValue = "toggle")
+                    String on
```
You can instead use a capital B `Boolean` which CAN be null, offer new default, then change the name to "enabled" and do a null-check. If "enabled" is null, toggle it, otherwise set debug to "enabled variable"

svdgoor replied:
That's indeed better. On it.

cyberpwnn commented on src/main/java/com/volmit/iris/core/decrees/DecObject.java:
```diff
@@ -302,4 +309,44 @@ public void setTile(int xx, int yy, int zz, TileData<? extends TileState> tile)
     }
 
+    //@Decree(description = "Paste a matter object", aliases = "matterpaste")
+    public void mpaste(
```
Delete this it will never be used

svdgoor replied:
ok

cyberpwnn commented on src/main/java/com/volmit/iris/core/decrees/DecObject.java:
```diff
@@ -302,4 +309,44 @@ public void setTile(int xx, int yy, int zz, TileData<? extends TileState> tile)
+            IrisDimension dimension,
+            @Param(description = "The file to store it in, can use / for subfolders")
+            String fileName,
+            @Param(description = "Overwrite existing object files", defaultValue = "false")
```
Add alias force

cyberpwnn commented on src/main/java/com/volmit/iris/core/decrees/DecObject.java:
```diff
@@ -302,4 +309,44 @@ public void setTile(int xx, int yy, int zz, TileData<? extends TileState> tile)
+            @Param(description = "The dimension to store the object in", contextual = true)
+            IrisDimension dimension,
+            @Param(description = "The file to store it in, can use / for subfolders")
+            String fileName,
```
change this to `name` not fileName

cyberpwnn commented on src/main/java/com/volmit/iris/util/decree/DecreeNode.java:
```diff
@@ -47,13 +47,21 @@ public DecreeNode(Object instance, Method method) {
      * @return The list of parameters if ALL are annotated by @{@link Param}, else null
      */
     public KList<DecreeParameter> getParameters() {
-        KList<DecreeParameter> p = new KList<>();
```
No

cyberpwnn commented on src/main/java/com/volmit/iris/util/decree/DecreeNode.java:
```diff
@@ -47,13 +47,21 @@ public DecreeNode(Object instance, Method method) {
     public KList<DecreeParameter> getParameters() {
-        KList<DecreeParameter> p = new KList<>();
+        KList<DecreeParameter> required = new KList<>();
+        KList<DecreeParameter> optional = new KList<>();
```
No, undo this entire file

svdgoor replied:
Why not put required params before non-required?

cyberpwnn commented on src/main/java/com/volmit/iris/util/decree/DecreeNode.java:
```diff
@@ -47,13 +47,21 @@ public DecreeNode(Object instance, Method method) {
+                optional.add(p);
+            }
         }
+        
```
no

svdgoor replied:
omfg

cyberpwnn commented on src/main/java/com/volmit/iris/util/decree/DecreeNode.java:
```diff
@@ -47,13 +47,21 @@ public DecreeNode(Object instance, Method method) {
+        
+        required.addAll(optional);
 
-        return p;
```
remove

cyberpwnn commented on src/main/java/com/volmit/iris/util/decree/DecreeNode.java:
```diff
@@ -47,13 +47,21 @@ public DecreeNode(Object instance, Method method) {
+        required.addAll(optional);
 
-        return p;
+        return required;
```
sthaaaaap

cyberpwnn commented on src/main/java/com/volmit/iris/util/plugin/VolmitSender.java:
```diff
@@ -423,55 +425,99 @@ public void sendDecreeHelp(VirtualDecreeCommand v) {
 
     public void sendDecreeHelpNode(VirtualDecreeCommand i){
         if (isPlayer()) {
```
Explain what you changed because it looks like it was just formatted, keep it the old way unless there's a good reason for it.

svdgoor replied:
Reformat only.
Behaves exactly the same except for a minor performance drop in exchange for actual readability and comprehensibility.

cyberpwnn commented on src/main/java/com/volmit/iris/util/plugin/VolmitSender.java:
```diff
@@ -423,55 +425,99 @@ public void sendDecreeHelp(VirtualDecreeCommand v) {
+                    .append(nodes);
+
+            sendMessageRaw(wrapper.toString());
+            System.out.println(wrapper);
```
Just undo this file too, but yeah no logging here

svdgoor committed:
Simple null check

svdgoor committed:
rename fileName + alias force

svdgoor committed:
Further simplifications of sender for clarity

svdgoor committed:
Merge remote-tracking branch 'upstream/master' into DecreeCommands

svdgoor committed:
doc

svdgoor commented:
- https://github.com/VolmitSoftware/Iris/pull/561#discussion_r691575112
- volmitSender command text refactor
Rest is resolved

svdgoor committed:
Shift

svdgoor committed:
Rework undo for Decree

Undo now in a service.
No longer per-player, collective history
We're not worledit - Dan
Removes the need for deprecated offlinePlayer calls as well.

svdgoor committed:
alias O in object because studiO was taking priority

svdgoor committed:
Merge remote-tracking branch 'upstream/master' into DecreeCommands

svdgoor committed:
Merge remote-tracking branch 'upstream/master' into DecreeCommands

svdgoor committed:
Merge remote-tracking branch 'upstream/master' into DecreeCommands

svdgoor committed:
Patch object handlers

svdgoor commented:
We need an IrisObjectPlacementScaleInterpolator handler
We need a fix for the object placement because it's placing null now.
```java.lang.Throwable: null
	at org.spigotmc.AsyncCatcher.catchOp(AsyncCatcher.java:15) ~[patched_1.17.1.jar:git-Purpur-1291]
	at net.minecraft.world.level.block.state.BlockBehaviour.onRemove(BlockBehaviour.java:145) ~[?:?]
	at net.minecraft.world.level.block.state.BlockBehaviour$BlockStateBase.onRemove(BlockBehaviour.java:914) ~[?:?]
	at net.minecraft.world.level.chunk.LevelChunk.setType(LevelChunk.java:676) ~[?:?]
	at net.minecraft.world.level.Level.setBlock(Level.java:640) ~[?:?]
	at net.minecraft.world.level.Level.setBlock(Level.java:602) ~[?:?]
	at org.bukkit.craftbukkit.v1_17_R1.block.CraftBlock.setTypeAndData(CraftBlock.java:199) ~[patched_1.17.1.jar:git-Purpur-1291]
	at org.bukkit.craftbukkit.v1_17_R1.block.CraftBlock.setBlockData(CraftBlock.java:184) ~[patched_1.17.1.jar:git-Purpur-1291]
	at org.bukkit.craftbukkit.v1_17_R1.block.CraftBlock.setBlockData(CraftBlock.java:178) ~[patched_1.17.1.jar:git-Purpur-1291]
	at com.volmit.iris.core.decrees.DecObject$1.set(DecObject.java:271) ~[?:?]
	at com.volmit.iris.engine.object.objects.IrisObject.place(IrisObject.java:648) ~[?:?]
	at com.volmit.iris.engine.object.objects.IrisObject.place(IrisObject.java:434) ~[?:?]
	at com.volmit.iris.core.decrees.DecObject.paste(DecObject.java:223) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
	at com.volmit.iris.util.decree.virtual.VirtualDecreeCommand.lambda$invokeNode$4(VirtualDecreeCommand.java:448) ~[?:?]
	at com.volmit.iris.util.decree.virtual.VirtualDecreeCommand.invokeNode(VirtualDecreeCommand.java:458) ~[?:?]
	at com.volmit.iris.util.decree.virtual.VirtualDecreeCommand.invoke(VirtualDecreeCommand.java:348) ~[?:?]
	at com.volmit.iris.util.decree.virtual.VirtualDecreeCommand.invoke(VirtualDecreeCommand.java:367) ~[?:?]
	at com.volmit.iris.util.decree.virtual.VirtualDecreeCommand.invoke(VirtualDecreeCommand.java:367) ~[?:?]
	at com.volmit.iris.util.decree.virtual.VirtualDecreeCommand.invoke(VirtualDecreeCommand.java:336) ~[?:?]
	at com.volmit.iris.util.decree.DecreeSystem.call(DecreeSystem.java:48) ~[?:?]
	at com.volmit.iris.util.decree.DecreeSystem.lambda$onCommand$1(DecreeSystem.java:64) ~[?:?]
	at org.bukkit.craftbukkit.v1_17_R1.scheduler.CraftTask.run(CraftTask.java:101) ~[patched_1.17.1.jar:git-Purpur-1291]
	at org.bukkit.craftbukkit.v1_17_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57) ~[patched_1.17.1.jar:git-Purpur-1291]
	at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[patched_1.17.1.jar:git-Purpur-1291]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[?:?]
	at java.lang.Thread.run(Thread.java:831) [?:?]```

cyberpwnn commented:
> We need an IrisObjectPlacementScaleInterpolator handler
> We need a fix for the object placement because it's placing null now.
> 
> ```
> 	at org.spigotmc.AsyncCatcher.catchOp(AsyncCatcher.java:15) ~[patched_1.17.1.jar:git-Purpur-1291]
> 	at net.minecraft.world.level.block.state.BlockBehaviour.onRemove(BlockBehaviour.java:145) ~[?:?]
> 	at net.minecraft.world.level.block.state.BlockBehaviour$BlockStateBase.onRemove(BlockBehaviour.java:914) ~[?:?]
> 	at net.minecraft.world.level.chunk.LevelChunk.setType(LevelChunk.java:676) ~[?:?]
> 	at net.minecraft.world.level.Level.setBlock(Level.java:640) ~[?:?]
> 	at net.minecraft.world.level.Level.setBlock(Level.java:602) ~[?:?]
> 	at org.bukkit.craftbukkit.v1_17_R1.block.CraftBlock.setTypeAndData(CraftBlock.java:199) ~[patched_1.17.1.jar:git-Purpur-1291]
> 	at org.bukkit.craftbukkit.v1_17_R1.block.CraftBlock.setBlockData(CraftBlock.java:184) ~[patched_1.17.1.jar:git-Purpur-1291]
> 	at org.bukkit.craftbukkit.v1_17_R1.block.CraftBlock.setBlockData(CraftBlock.java:178) ~[patched_1.17.1.jar:git-Purpur-1291]
> 	at com.volmit.iris.core.decrees.DecObject$1.set(DecObject.java:271) ~[?:?]
> 	at com.volmit.iris.engine.object.objects.IrisObject.place(IrisObject.java:648) ~[?:?]
> 	at com.volmit.iris.engine.object.objects.IrisObject.place(IrisObject.java:434) ~[?:?]
> 	at com.volmit.iris.core.decrees.DecObject.paste(DecObject.java:223) ~[?:?]
> 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
> 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[?:?]
> 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
> 	at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
> 	at com.volmit.iris.util.decree.virtual.VirtualDecreeCommand.lambda$invokeNode$4(VirtualDecreeCommand.java:448) ~[?:?]
> 	at com.volmit.iris.util.decree.virtual.VirtualDecreeCommand.invokeNode(VirtualDecreeCommand.java:458) ~[?:?]
> 	at com.volmit.iris.util.decree.virtual.VirtualDecreeCommand.invoke(VirtualDecreeCommand.java:348) ~[?:?]
> 	at com.volmit.iris.util.decree.virtual.VirtualDecreeCommand.invoke(VirtualDecreeCommand.java:367) ~[?:?]
> 	at com.volmit.iris.util.decree.virtual.VirtualDecreeCommand.invoke(VirtualDecreeCommand.java:367) ~[?:?]
> 	at com.volmit.iris.util.decree.virtual.VirtualDecreeCommand.invoke(VirtualDecreeCommand.java:336) ~[?:?]
> 	at com.volmit.iris.util.decree.DecreeSystem.call(DecreeSystem.java:48) ~[?:?]
> 	at com.volmit.iris.util.decree.DecreeSystem.lambda$onCommand$1(DecreeSystem.java:64) ~[?:?]
> 	at org.bukkit.craftbukkit.v1_17_R1.scheduler.CraftTask.run(CraftTask.java:101) ~[patched_1.17.1.jar:git-Purpur-1291]
> 	at org.bukkit.craftbukkit.v1_17_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57) ~[patched_1.17.1.jar:git-Purpur-1291]
> 	at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[patched_1.17.1.jar:git-Purpur-1291]
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[?:?]
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[?:?]
> 	at java.lang.Thread.run(Thread.java:831) [?:?]```
> ```

Stop placing in the minecraft world async

svdgoor commented:
Ok

svdgoor committed:
Merge remote-tracking branch 'upstream/master' into DecreeCommands

svdgoor committed:
Merge remote-tracking branch 'upstream/master' into DecreeCommands

svdgoor committed:
Merge remote-tracking branch 'upstream/master' into DecreeCommands

svdgoor committed:
Auto stash before merge of "DecreeCommands" and "upstream/master"

svdgoor committed:
Merge remote-tracking branch 'upstream/master' into DecreeCommands

svdgoor committed:
Merge remote-tracking branch 'upstream/master' into DecreeCommands

svdgoor committed:
What & debug fix

svdgoor committed:
Merge remote-tracking branch 'upstream/master' into DecreeCommands

svdgoor committed:
Merge remote-tracking branch 'upstream/master' into DecreeCommands

svdgoor committed:
Update Commands with patches

svdgoor committed:
remove what
